lib/datastax/graph/result-set.js

```diff
@@ -1,141 +1,56 @@
 "use strict";
 
-const utils = require("../../utils");
+const utils = require("../utils");
 
 /**
- * Creates a new instance of <code>GraphResultSet</code>.
- * @class
- * @classdesc
- * Represents the result set of a [graph query execution]{@link Client#executeGraph} containing vertices, edges or
- * scalar values depending on the query.
- * <p>
- * It allows iteration of the items using <code>for..of</code> statements under ES2015 and exposes
- * <code>forEach()</code>, <code>first()</code> and <code>toArray()</code> to access the underlying items.
- * </p>
- * @example
- * for (let vertex of result} { ... }
- * @example
- * const arr = result.toArray();
- * @example
- * const vertex = result.first();
- * @param {ResultSet} result
- * @param {Function} [rowParser]
- * @alias module:datastax/graph~GraphResultSet
- * @constructor
+ * @deprecated
  */
 function GraphResultSet(result, rowParser) {
+  throw new ReferenceError(utils.markAsNotSupported("class", "GraphResultSet"));
```

Comment:
I think we also considered adding some way of stoping typescript compilation from finishing. Did you considered that / checked it would work as expected?

Reply:
(Work as expected: would trigger compilation error only if used)

Reply:
Right, I forgot about this requirement. Let's ensure it's true, if it's doable.